test/integration/config-validation/test/index.test.ts

```diff
@@ -54,5 +54,28 @@ describe('next.config.js validation', () => {
         }
       }
     )
+
+    it('should allow undefined environment variables', async () => {
+      const configContent = `
+        module.exports = {
+          env: {
+            FOO: 'bar',
+            QUX: undefined
+          }
+        }
+      `
+
+      await fs.writeFile(nextConfigPath, configContent)
+      const result = await nextBuild(path.join(__dirname, '../'), undefined, {
+        stderr: true,
+        stdout: true,
+      })
+
+      await fs.remove(nextConfigPath)
+
+      expect(result.stdout + result.stderr).not.toContain(
+        '"env.QUX" is missing, expected string'
```

Comment:
If the expected type ever changes in the future string might not be sufficient but the test wouldn't fail, so changing it to something a bit more generic would catch that case:
```suggestion
        '"env.QUX" is missing'
```